packages/react-form-with-layout/src/index.js

```diff
@@ -57,46 +57,44 @@ export default class FormWithLayout extends React.Component {
   }
 
   renderField(name) {
-    const props = this.props;
-
-    const value = inputValueLookup(props.values || {}, name);
-    const error = props.errors ? props.errors[name] : null;
-    const def = props.getFieldProps(name);
-
-    // form.defaultValues(field).defaultValue
-    var defaultValue = inputValueLookup(props.defaultValues || {}, name);
-
-    // fieldProps.value > form.defaultValues(field).defaultValue
-    defaultValue =
-      def.defaultValue === null || typeof def.defaultValue === 'undefined'
-        ? defaultValue
-        : def.defaultValue;
-
-    // form.props.disabled > fieldProps.disabled
-    var disabled = props.disabled || def.disabled;
-
-    var label = def.label ? def.label : humanize(name);
-
-    var refName = this.getFieldRef(name);
-
-    const fieldProps = assign({}, def, {
-      ref: c => {
-        this.fields[refName] = c;
-      },
-      // key: refName,
+    const formProps = this.props;
+    const fieldProps = formProps.getFieldProps(name);
+
+    const label = typeof fieldProps.label !== 'undefined'
+      ? fieldProps.label
+      : humanize(name);
+    const refName = this.getFieldRef(name);
+
+    // formProps.disabled > fieldProps.disabled
+    const disabled = typeof formProps.disabled !== 'undefined'
+      ? formProps.disabled
+      : fieldProps.disabled;
+
+    // fieldProps.defaultValue > formProps.defaultValues(field).defaultValue
+    const defaultValue = typeof fieldProps.defaultValue !== 'undefined'
+        ? fieldProps.defaultValue
+        : inputValueLookup(formProps.defaultValues || {}, name);
+
+    // fieldProps.value > formProps.values(field).value
+    const value = typeof fieldProps.value !== 'undefined'
+        ? fieldProps.value
+        : inputValueLookup(formProps.values || {}, name);
+
+    const inputProps = assign({}, fieldProps, {
+      ref: c => this.fields[refName] = c,
       className: 'field',
-      field: def,
       name: name,
       value: value,
-      defaultValue: defaultValue,
       label: label,
-      error: error,
-      disabled: disabled
+      disabled: disabled,
+
+      defaultValue: typeof value !== 'undefined' ? undefined : defaultValue,
```

Comment:
if there is `value`, `defaultValue` should not be used, this will cause a react warning/error complaining about both should not exist at the same time